packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+        getTestNode(dataStore.id).parents.delete(parentDataStore.id);
+        updateUnreferencedDataStores(dataStore.id);
+    };
+
+    const getRandomContainer = (): IContainer => {
+        const id = randomInteger(0, connectedContainers.length - 1);
+        return connectedContainers[id];
+    };
+
+    const getRandomDataStoreFromContainer = async (container: IContainer) => {
+        const random = randomInteger(0, testNodes.size - 1);
+        const dataStoreId = Array.from(testNodes.values())[random].id;
+        return requestFluidObject<TestDataObject>(container, dataStoreId);
+    };
+
+    const getRandomParentForDataStore = async (dataStore: TestDataObject, container: IContainer) => {
+        const parents = getTestNode(dataStore.id).parents;
+        const random = randomInteger(0, parents.size - 1);
+        const parentId = Array.from(parents.values())[random];
+        return requestFluidObject<TestDataObject>(container, parentId);
+    };
+
+    const referenceRandomDataStore = async () => {
+        if (connectedContainers.length <= 0 || testNodes.size <= 0) {
+            return;
+        }
+        const container = getRandomContainer();
+        const dataStore = await getRandomDataStoreFromContainer(container);
+        const parent = await getRandomDataStoreFromContainer(container);
+        referenceDataStore(dataStore, parent);
+    };
+
+    const unreferenceRandomDataStore = async () => {
+        if (connectedContainers.length <= 0 || testNodes.size <= 0) {
+            return;
+        }
+        const container = getRandomContainer();
+        const dataStore = await getRandomDataStoreFromContainer(container);
+        const parentDataStore = await getRandomParentForDataStore(dataStore, container);
+        unreferenceDataStore(dataStore, parentDataStore);
+    };
+
+    const actionsList: (() => Promise<unknown>)[] = [
+        loadNewContainer,
+        closeRandomContainer,
+        createDataStoreForRandomContainer,
+        referenceRandomDataStore,
+        unreferenceRandomDataStore,
+    ];
+    let testTime = 0;
+    const testEnd = 60 * 1000; // 1 minute
+    const testActions: { sleep: number; action: () => Promise<unknown>; }[] = [];
+    while (testTime < testEnd) {
+        const sleepTime: number = randomInteger(0, defaultSessionExpiryDurationMs + 1000);
+        const action = makeRandom().pick(actionsList);
+        testActions.push({ sleep: sleepTime, action });
+        testTime += sleepTime;
+    }
+
+    it("Create and reference and unreference datastores with multiple containers", async () => {
+        overrideLogger.ignoreExpectedEventTypes({
+            eventName: "fluid:telemetry:Container:ContainerClose",
+            errorType: ContainerErrorType.clientSessionExpiredError,
+        });
+
+        for (const action of testActions) {
+            void action.action();
+            await sleep(action.sleep);
+        }
```

Comment:
I was definitely expecting to see some code at some point inspect the GC Data and compare to the TestNodes / ref/unref lists.  Is that coming in a later change?  Consider reverting all the ref/unref stuff for now if you're not going to use it (I know it's used when referencing or unreferencing a DataStore, but is it necessary? Probably, but if not then this first PR can be made smaller/more scoped)